### ISSUE TITLE
Get Conversations doesn't work

### ISSUE THREAD

**Issue by bandtank** [MicrosoftDocs/bot-docs#2266]
https://learn.microsoft.com/en-us/azure/bot-service/rest-api/bot-framework-rest-connector-api-reference?view=azure-bot-service-4.0#get-conversations

This endpoint does not work as explained in the documentation.

![image](https://user-images.githubusercontent.com/1170415/216794022-70ca38ac-3e07-433a-9596-30f672d9d361.png)

There is also no explanation of the `continuationToken` parameter or how to find it.
---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: c36ddb50-9e44-96e3-2df3-174166efa75d
* Version Independent ID: 68ac4b64-6616-2216-3519-e0b33d3a53e4
* Content: [Bot Framework Connector service REST API reference - Bot Service](https://learn.microsoft.com/en-us/azure/bot-service/rest-api/bot-framework-rest-connector-api-reference?view=azure-bot-service-4.0#get-conversations)
* Content Source: [articles/rest-api/bot-framework-rest-connector-api-reference.md](https://github.com/MicrosoftDocs/bot-docs-pr/blob/live/articles/rest-api/bot-framework-rest-connector-api-reference.md)
* Service: **bot-service**
* GitHub Login: @JonathanFingold
* Microsoft Alias: **iawilt**

**Comment by timenick** [comment]
ContinuationToken is a property of ConversationsResult Object could be found here https://learn.microsoft.com/en-us/azure/bot-service/rest-api/bot-framework-rest-connector-api-reference?view=azure-bot-service-4.0#conversationsresult-object

For detailed explanation you could refer to the source code on Github here https://github.com/microsoft/botbuilder-dotnet/blob/731e157b9a91aa13c3359a01ba0c7fe0a581a787/libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs#L756

**Comment by bandtank** [comment]
Sorry, but that was not helpful. The documentation says `GET /v3/conversations` should work, but it doesn't.

```
GET /amer/v3/conversations HTTP/1.1
Authorization: Bearer eyJ0eXAiO...
Host: smba.trafficmanager.net
 
HTTP/1.1 405 Method Not Allowed
Allow: POST
Content-Length: 72
Content-Type: application/json; charset=utf-8
Server: Microsoft-HTTPAPI/2.0
Date: Tue, 07 Feb 2023 11:55:14 GMT
 
{"message":"The requested resource does not support http method 'GET'."}
```

**Comment by hibrenda** [comment]
@bandtank  Does above comment solve your issue ?

**Comment by bandtank** [comment]
No. The primary issue is getting a list of conversations, which has not been answered.

**Comment by hibrenda** [comment]
@bandtank 

Could you try following ? 
- baseURL is https://smba.trafficmanager.net/teams/
- Continuetoken should be empty for the first-time. and then this function Lists the Conversations in which this bot has participated for a given channel server. The channel server returns results in pages and each page will include a `continuationToken` that can be used to fetch the next page of results from the server.

If this does not work, let us know what error you got.

**Comment by timenick** [comment]
The Teams channel does not seem to support this method. Response body says it all.

**Comment by bandtank** [comment]
It doesn't say that. It says the `GET` method is not supported, which is not the same. The error message should be clarified if the intent is to disallow the endpoint.

If the endpoint doesn't work through Teams, how can I get a list of conversations?

**Comment by hibrenda** [comment]
@bandtank   Teams do not support the get conversation API. When the bot is installed in any content it will get an installationUpdate event with the conversation in which it was added. If it needs that information, it can store it when the event is fired. Every message/event the bot receive will also have the context of the conversation in which it happened.  Will this help ?

**Comment by bandtank** [comment]
@hibrenda What happens if the server isn't responding or the message is missed? The user might not message the bot again, so how would you get the conversation ID? This is an oversight in the Teams messaging system. There are many situations that may occur which would prevent the bot from gaining access to the user's bot ID and personal conversation ID. This endpoint needs to exist in Teams or an equivalent endpoint needs to be exposed.

**Comment by hibrenda** [comment]
@bandtank  Thanks for your input. I've passed this request to Teams. They will note in the backlog.

**Comment by lorieide** [comment]
#reassign:@iaanw
